lambdas/db-migration/main.tf

```diff
@@ -81,7 +84,8 @@ resource "aws_lambda_function" "db_migration" {
     content {
       subnet_ids = var.subnet_ids
       security_group_ids = [
-        aws_security_group.db_migration[0].id
+        aws_security_group.db_migration[0].id,
```

Comment:
Not really related to your work but I was wondering why this Lambda is creating its own security group at all?

Reply:
So - 

This module assumes a VPC (is that valid now?   Other modules do not.... hmm)  - it in theory needs a security group for egress *if the database isn't in the VPC*.... but I expect the intention here was just that it needs a security group. 

So probably related to this work to remove that and just use the provided DB group if we're willing to concede that database work requires operations inside a VPC.   Or *really* if users want to use a non-VPC database setup they could pass in a SG that sets all egress or something. 

🤔

Reply:
This raises another question re: if the networking settings in Core make sense around this for this work, given we're only using the DB security group if VPC is enabled and subnets are passed in.

Reply:
Are you saying that the VPC/subnet settings for the DB migration should be required, not optional? It seems like it really depends if the use of the security group for RDS is optional? And if it isn't, why not?

Reply:
No - I'm wondering if the SG for the database should be optional in Core.  

A VPC security group only works within a VPC - if you're running outside of one, I believe you need a DB security group with IP ranges, etc if you're trying to provide access to a RDS instance outside of a VPC.  

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.RDSSecurityGroups.html

tl;dr case 1) 

DB/lambda are in a VPC.   In this instance you can just put both an an accessor security group.  

2) DB or lambda not in a VPC    You then need a security group with rules specific to your access patterns, and the lambda has to be configured to allow egress to the <range the DB will deploy to - I'm a bit fuzzy on this> 


Case #1 makes obvious sense for most internal user use cases.      Case 2 might make sense if you were leveraging an existing postgres/rds database or some other more complicated network topology.

In thinking about this, the Core configuration concern is moot, as a lambda not in a VPC already has external access and can't manage DB access via a security group anyway.

Reply:
Yeah let's discuss this offline

Reply:
@markdboyd  re-commenting so it gets on the agenda

Reply:
Discussed this and we're good. This code supports having an RDS cluster deployed inside and outside of a VPC